code-challenges.test.js

```diff
@@ -21,30 +20,133 @@ const colors1 = ["purple", "blue", "green", "yellow", "pink"]
+  arr.shift();
+  // Return the new, shuffled array using .sort() and Math.random() and return the array.
+  // Because the values inside the arrays are strings we can use .sort(), else if it was a number, it will convert it to a string. 
+  arr.sort((a, b) => 0.5 - Math.random())
+  //return the array.
+  return arr;
+};
 
 // --------------------2) Create a function that takes in an object that contains up votes and down votes and returns the net total of votes.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const votes1 = {upVotes: 13, downVotes: 2}
-// Expected output: 11
-const votes2 = {upVotes: 2, downVotes: 33}
+// const votes1 = {upVotes: 13, downVotes: 2}
+// // Expected output: 11
+// const votes2 = {upVotes: 2, downVotes: 33}
 // Expected output: -31
 
 
+describe("addVotes", () => {
+    it("takes in an object that contains up votes and down votes and returns the net total of votes.", () => {
+
+        const votes1 = {
+            upVotes: 13, 
+            downVotes: 2
+        };
+
+      expect(addVotes(votes1)).toEqual(
+        (11));
+
+        const votes2 = {
+            upVotes: 2, 
+            downVotes: 33
+        };
+
+      expect(addVotes(votes2)).toEqual(
+        (-31));
+    });
+  });
+
+  // output: ReferenceError: addVotes is not defined (Good Fail)
+
 // b) Create the function that makes the test pass.
+// declare function, call it 'addVotes' that takes in an object.
+// and returns the object upVote value - the object downVotes to get the net total of votes.
+const addVotes = (object) => {
+    return object.upVotes - object.downVotes
+  }
 
 
 // --------------------3) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator to pass in a dynamic number of arguments.
 
 // a) Create a test with an expect statement using the variables provided.
 
-const dataArray1 = ["array", "object", "number", "string", "Boolean"]
-const dataArray2 = ["string", "null", "Boolean", "string", "undefined"]
+// const dataArray1 = ["array", "object", "number", "string", "Boolean"]
+// const dataArray2 = ["string", "null", "Boolean", "string", "undefined"]
 // Expected output: ["array", "object", "number", "string", "Boolean", "null", "undefined"]
 
-
-// b) Create the function that makes the test pass.
+describe("removeDuplicates", () => {
+    it("takes in two arrays as arguments and returns one array with no duplicate", () => {
+      
+      const dataArray1 = ["array", "object", "number", "string", "Boolean"];
+      const dataArray2 = ["string", "null", "Boolean", "string", "undefined"];
+
+      expect(removeDuplicates(dataArray1, dataArray2)).toEqual(
+        expect.arrayContaining([
+          "array",
+          "object",
+          "number",
+          "string",
+          "Boolean",
+          "null",
+          "undefined",
+        ])
```

Comment:
Careful on the matcher you are using.  The expect.arrayContaining matcher should only be used when you know you need a variation or the array but may not return a predictable result.  In this problem, the result should be predictable, so I would recommend sticking with just .toEqual.